SkyEditor.RomEditor.Rtdx/Domain/Rtdx/Structures/CodeTable.cs

```diff
@@ -1,13 +1,31 @@
+using System;
 using System.Collections.Generic;
 using System.Text;
+using System.Text.RegularExpressions;
 using SkyEditor.IO.Binary;
 using SkyEditor.RomEditor.Domain.Common.Structures;
 
 namespace SkyEditor.RomEditor.Domain.Rtdx.Structures
 {
+    /// <summary>
+    /// This class allows encoding and decoding strings between human-readable text directives like [hero]
+    /// and an internal representation with special Unicode characters.
+    /// </summary>
+    /// Note that the code_table.bin file is used, which doesn't always match the directives that are used
+    /// in the Japanese text inside the .lua files. It seems like the game uses either a combination of hard-coded
+    /// regular expressions and code_table.bin to parse these or even hardcodes everything and ignores code_table.bin
+    /// completely. However, this should work for assembling custom text with the most common directives
+    /// like [hero], [partner] etc.
     public class CodeTable
     {
-        public List<Entry> Entries { get; } = new List<Entry>();
+        public Dictionary<ushort, Entry> EntriesByUnicode { get; } = new Dictionary<ushort, Entry>();
+        public Dictionary<string, Entry> EntriesByCodeString { get; } = new Dictionary<string, Entry>();
+
+        private static Regex StringTokenRegex = new Regex("\\[(.+?:?)(\\d+)*\\]");
```

Comment:
private static readonly

Reply:
Can also tack on a `, RegexOptions.Compiled);` at the end to speed up continual uses of it.